lib/utils.ts

```diff
@@ -861,47 +983,61 @@ export function getMessagesNode(ruleInfo, scopeManager) {
 
 /**
  * Get the list of messageId properties from `meta.messages` for a rule.
- * @param {RuleInfo} ruleInfo
- * @param {ScopeManager} scopeManager
- * @returns {Node[]|undefined}
+ * @param ruleInfo
+ * @param scopeManager
  */
-export function getMessageIdNodes(ruleInfo, scopeManager) {
+export function getMessageIdNodes(
+  ruleInfo: RuleInfo,
+  scopeManager: Scope.ScopeManager,
+): (Property | SpreadElement)[] | undefined {
   const messagesNode = getMessagesNode(ruleInfo, scopeManager);
 
   return messagesNode && messagesNode.type === 'ObjectExpression'
     ? evaluateObjectProperties(messagesNode, scopeManager)
     : undefined;
 }
 
+const isProperty = (node: Node): node is Property => node.type === 'Property';
```

Comment:
I don't think having this helper is necessary?

Reply:
Updated